tests/tasks/setup_candlepin.yml

```diff
@@ -0,0 +1,89 @@
+---
+- name: Get LSR_RHC_TEST_DATA environment variable
+  set_fact:
+    lsr_rhc_test_data_file: "{{ lookup('env', 'LSR_RHC_TEST_DATA') }}"
+  delegate_to: 127.0.0.1
+  become: false
+
+- name: Import test data
+  include_vars:
+    file: "{{ lsr_rhc_test_data_file }}"
+  when: lsr_rhc_test_data_file | length > 0
+  delegate_to: 127.0.0.1
+  become: false
+
+- name: Deploy Candlepin
+  block:
+    - name: Clone ansible-role-candlepin repo
+      git:
+        repo: https://github.com/candlepin/ansible-role-candlepin.git
+        version: master
+        dest: /tmp/candlepin-role-repo
+      delegate_to: 127.0.0.1
+      become: false
```

Comment:
This will likely conflict if one or more tests are run at the same time on the same control node (i.e. localhost i.e. 127.0.0.1) which our CI system does.   Instead, use the `tempfile` module to create a per-test temporary working directory: https://docs.ansible.com/ansible/2.9/modules/tempfile_module.html#tempfile-module
```
  block:
    - name: Create working directory for test
      tempfile:
        state: directory
        prefix: lsr_
        suffix: _rhc
      register: __rhc_test_dir
      delegate_to: 127.0.0.1
      become: false

    - name: Clone ansible-role-candlepin repo
      git:
        repo: https://github.com/candlepin/ansible-role-candlepin.git
        version: master
        dest: "{{ __rhc_test_dir.path }}/role-repo"
      delegate_to: 127.0.0.1
      become: false
```

Reply:
hm TBH i noticed the same approach as used in the `certificates` role for deploying freeipa; another possibility i thought about was to avoid the cloning in `/tmp` + symlink and directly clone under `tests/roles/` (which i guess has the same issue)

maybe i'm missing something: if i create and use a temporary directory as you shown, how is that used by ansible automatically?

Reply:
> hm TBH i noticed the same approach as used in the `certificates` role for deploying freeipa; another possibility i thought about was to avoid the cloning in `/tmp` + symlink and directly clone under `tests/roles/` (which i guess has the same issue)

There's only 1 test that does that in the certificate role, so there is no chance of conflict if multiple tests are run in parallel.

> 
> maybe i'm missing something: if i create and use a temporary directory as you shown, how is that used by ansible automatically?

It isn't - that is, if you use `include_role`, I'm not sure how you tell it to look in the temporary directory for the role + collections.  Another way to do this would be to use `command: ansible-playbook ....` to install candlepin, and create a short playbook file in the temp directory.  I need to think about this some more.

Reply:
I think what you have here is ok for now.  Maybe just a note to developers to tell them to clean up their ~/.ansible directory after running the test to remove the collections installed by the test.

Reply:
> Maybe just a note to developers to tell them to clean up their ~/.ansible directory after running the test to remove the collections installed by the test.

Hm with my latest changes I don't have anything installed locally in my `~/.ansible/` by the tests, everything is in `.tox/`. Do you get anything in your ~?